src/main/java/org/opensearch/sdk/handlers/ExtensionsRestRequestHandler.java

```diff
@@ -68,11 +73,19 @@ public RestExecuteOnExtensionResponse handleRestExecuteOnExtensionRequest(Extens
             );
         }
 
+        String oboToken = request.getRequestIssuerIdentity();
```

Comment:
Seems like this should be called an onBehalfOfToken both here and on the request object?

What do you think about making the type `Optional<>` ?

Reply:
Agreed. I think this needs to be updated in core. I can create a PR for that.